docs/extending/creating_custom_dropdowns.md

```diff
@@ -5,83 +5,140 @@ Follow the steps below to learn how to integrate this component to fit it to you
+If you need a multilevel `choices`, use the following structure:
+
+```twig
+{% set choices = [{
+    value: "DE",
+    label: "DE"
+}, {
+    value: "US",
+    label: "US"
+}, {
+    value: "NO",
+    label: "NO"
+}, {
+    value: "PL",
+    label: "PL",
+    choices: [{
+        value: "PL_S",
+        label: "Silesian"
+    }, {
+        value: "PL_K",
+        label: "Kashubian"
+    }]
+}] %}
+```
 
-You can provide placeholder text for your custom drop-down. To do so:
+To set `preferred_choices`, use the following:
 
-- put a `data-value` attribute with no value `data-value=""`
-- add a `disabled` attribute to the item in the duplicated list of options to make it unclickable.
+```twig
+{% set preferred_choices = [{
+    value: "NO",
+    label: "NO"
+}] %}
+```
 
+For `value`, see the example:
 
-Example:  
- 
-```html
-<li data-value="" class="ibexa-custom-dropdown__item" disabled>Select an option</li>
-<li data-value="1" class="ibexa-custom-dropdown__item">Option 1</li>
+```twig
+{% set value = [{
+    value: "DE",
+    label: "DE"
+}] %}
 ```
 
+You can configure the following attributes:
+
+Name|Values|Definition|
+|---|------|----------|
+|`source`| - |What is currently defined in the `<select>` input header.|
+|`choices`| - |Elements listed in the drop-down.|
+|`preferred_choices`|Elements listed at the top of the list with a separator.|
+|`value`|-|The currently selected element.It is an object with a key `value`. |
+|`multiple`| true</br>false|Boolean. To allow users to select multiple items, add this attribute to the same element.|
+|`translation_domain`|true</br>false|Used for translating choices and placeholder.|
+|`custom_form`|true</br>false|For custom form must be set to true.|
+|`class`| - |Additional classes for the element with `ibexa-dropdown` class.|
+`placeholder`|Displayed placeholder when no option is selected.|
+|`custom_init`|true</br>false|By default set to `false`. If set to `true`, requires to manually initialize drop-down in the JavaScript.|
+|`is_disabled`|true</br>false|Disables drop-down.|
+|`is_hidden`|true</br>false|Hides the whole widget.|
+|`is_small`|true</br>false|Adjusts height of the widget (from 48px to 32px).|
+|`is_ghost`|true</br>false|Changes layout of the widget, removes all borders and backgrounds (similar to buttons modifier).|
+|`min_search_items`|number, default 5|Minimum number of options that has to be passed to show the search inside the drop-down.|
+|`selected_item_label`|text|Allows to set constant label for widget, by default visible label shows currently selected options.|
+
+![Drop-down expanded state](img/dropdown_expanded_state.png)
+
 ## Initialize
 
-To initialize a custom drop-down, run the following JavaScript code:
+All drop-downs are searched and initialized automatically in `admin.dropdown.js`. To extend or modify it, you need to add a `custom_init` parameter to twig element. otherwise it will be initialized two times. Next, run the following JavaScript code:
```

Comment:
```suggestion
All drop-downs are searched and initialized automatically in `admin.dropdown.js`.
To extend or modify the search, you need to add a `custom_init` parameter to the Twig element. Otherwise it will be initialized two times.
Next, run the following JavaScript code:
```
"extend or modify the search", or is something else meant?